File: 프로그래머스/lv2/권민서/숫자의표현.java
import java.util.*;

class Solution {

    public int solution(int n) {

        int lt = 1;
        int rt = 1;
        int tmp = 0;
        int answer = 1;

        if(n < 3) return 1;

        while (lt < n) {

            if(tmp == n) {
                answer++;
            }
            if (tmp < n) {
                tmp += rt++;
                continue;
            }
            tmp -= lt++;
        }
        return answer;
    }
}

File: 프로그래머스/lv2/김미래/숫자의 표현.cpp
#include <string>
#include <vector>

using namespace std;


int solution(int n) {
    int answer = 0;
    int start = 1, finish = 1;
    int sum = 1;

    while (start <= n) {
        if (sum < n) {
            finish++;
            sum += finish;
        } else if (sum > n) {
            sum -= start;
            start++;
        } else { 
            answer++;
            sum -= start;
            start++;
        }
    }
    
    return answer;
}
